README.MD

```diff
@@ -0,0 +1,326 @@
+});
+
+
+```   
+
+##### Deploy BrandedToken contract
+BrandedToken is a value-backed EIP-20 token with a fixed conversion rate against the Value Token chosen. This contract is deployed on the value chain.
+```js
+const brandedTokenHelperConfig = {
+    deployer: deployerAddress,
+    valueToken: eip20Token,
+    symbol: 'BT',
+    name: 'MyBrandedToken',
+    decimals: '18',
+    conversionRate: '10',
+    conversionRateDecimals: 5,
+    organization: organizationAddress
+};
+deployParams = {
+  gas: '7500000',
+  gasPrice: '0x3B9ACA00'
+};
+const brandedTokenHelper = Setup.BrandedTokenHelper;
+let brandedTokenAddress;
+brandedTokenHelper.setup(brandedTokenHelperConfig, deployParams).then(function(){
+    brandedTokenAddress = brandedTokenHelper.address;
+});
+
+
+```  
+
+##### Deploy UtilityBrandedToken contract
+The UtilityBrandedToken is a representation of the BrandedToken on the sidechains. Thus, this contract is deployed on the sidechain
+
+```js
+const utilityBrandedTokenHelperConfig = {
+    deployer: deployerAddress,
+    token: brandedTokenAddress,
+    symbol: "UBT",
+    name: "MyBrandedToken",
+    decimals: "18",
+    organization: organizationAddress
+};
+deployParams = {
+  gas: '7500000',
+  gasPrice: '0x3B9ACA00'
+};
+const utilityBrandedTokenHelper = Setup.UtilityBrandedTokenHelper;
+let utilityBrandedTokenAddress;
+utilityBrandedTokenHelper.setup(utilityBrandedTokenHelperConfig, deployParams).then(function(){
+    utilityBrandedTokenAddress = utilityBrandedTokenHelper.address;
+});
+
+
+```
+
+##### Deploy GatewayComposer contract
+Gateway composer is a contract optimizes the transactions required to perform the stake and mint process. This contract is deployed on the value chain.
+
+```js
+const gatewayComposerConfig = {
+    deployer: deployerAddress,
+    owner: staker,
+    valueToken: eip20Token,
+    brandedToken: brandedTokenAddress
+};
+deployParams = {
+  gas: '7500000',
+  gasPrice: '0x3B9ACA00'
+};
+const gatewayComposerHelper = Setup.GatewayComposerHelper;
+let gatewayComposerAddress;
+gatewayComposerHelper.setup(gatewayComposerConfig, deployParams).then(function(){
+    gatewayComposerAddress = gatewayComposerHelper.address;
+});
+
+```
+
+##### Deploy Gateway Mock contract
+Gateway contract facilitates the transposition of the BrandedToken from the value chain to the sidechain. The contract is deployed on the value chain.
+
+```js
+let gatewayAddress;
+mockContractDeployerObject.deployMockGatewayPass().then(function(){
+      gatewayAddress = mockContractDeployerObject.addresses.MockGatewayPass;
+});
+
+```
+
+#### BrandedToken Staking through GatewayComposer
+
+##### Approve GatewayComposer for ValueToken & Calls requestStake
+
+```js
+const StakeHelper = BrandedToken.Helpers.StakeHelper;
+const stakeHelperInstance = new StakeHelper(web3Provider, brandedTokenAddress, gatewayComposerAddress);
+let txOptions = {
+  from: owner
+};
+const stakeAmountInWei = '100';
+const stakerGatewayNonce = 1;
```

Comment:
This requires an explanation.

Reply:
@abhayks1

Reply:
Update with below:
```
// stakerGatewayNonce is the nonce of the staker address stored in Gateway contract. Alternatively it can also be fetched with Gateway.getNonce(stakeAddress) method.
const stakerGatewayNonce = 1;
```